packages/office-ui-fabric-react/src/components/SearchBox/SearchBox.base.tsx

```diff
@@ -201,30 +201,23 @@ export class SearchBoxBase extends React.Component<ISearchBoxProps, ISearchBoxSt
     switch (ev.which) {
       case KeyCodes.escape:
         this.props.onEscape && this.props.onEscape(ev);
-        if (!ev.defaultPrevented) {
+        if (this.state.value && !ev.defaultPrevented) {
```

Comment:
Can you add a comment about why we don't call `onClear` when there's not a value?